utilities/src/offchain_worker/iterator_helper.rs

```diff
@@ -61,89 +215,144 @@ impl<T> Iterator for MapIterator<T> {
 	}
 }
 
-/// A strongly-typed double map in storage whose secondary keys and values can
-/// be iterated over.
-pub trait RandomIterableStorageDoubleMap<K1: FullCodec, K2: FullCodec, V: FullCodec>:
-	StorageDoubleMap<K1, K2, V>
-{
-	/// The type that iterates over all `(key2, value)`.
-	type PrefixIterator: Iterator<Item = (K2, V)>;
+/// Shim for MapIterator, add more features
+pub struct MapIteratorShim<'a, 'b, T, L: Lockable> {
+	pub map_iterator: MapIterator<T>,
+	pub remain_iterator_count: Option<u32>,
+	pub lock_guard: Option<StorageLockGuard<'a, 'b, L>>,
```

Comment:
do we need to pass this? can we do the extend lock in the code that using this iterator?
this prevent us using this iterator within wasm runtime. we don't have a use case yet but I don't see why we need to prevent this possibility.